pkg/csi/service/vanilla/controller.go

```diff
@@ -515,13 +520,35 @@ func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolume
 		ScParams:   scParams,
 		VolumeType: common.FileVolumeType,
 	}
-
-	volumeID, err := common.CreateFileVolumeUtil(ctx, cnstypes.CnsClusterFlavorVanilla, c.manager, &createVolumeSpec)
-	if err != nil {
-		msg := fmt.Sprintf("failed to create volume. Error: %+v", err)
-		log.Error(msg)
-		return nil, status.Errorf(codes.Internal, msg)
+	var volumeID string
+	if containerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIAuthCheck) {
+		dsURLToInfoMap := c.authMgr.GetDatastoreMapForFileVolumes(ctx)
+		log.Debugf("Filtered Datastores: %+v", dsURLToInfoMap)
+		var filteredDatastores []*cnsvsphere.DatastoreInfo
+		for _, datastore := range dsURLToInfoMap {
+			filteredDatastores = append(filteredDatastores, datastore)
+		}
+		if len(filteredDatastores) == 0 {
+			msg := "No datastores found to create file volume"
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		volumeID, err = common.CreateFileVolumeUtil(ctx, cnstypes.CnsClusterFlavorVanilla,
+			c.manager, &createVolumeSpec, filteredDatastores)
+		if err != nil {
+			msg := fmt.Sprintf("failed to create volume. Error: %+v", err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+	} else {
+		volumeID, err = common.CreateFileVolumeUtilOld(ctx, cnstypes.CnsClusterFlavorVanilla, c.manager, &createVolumeSpec)
```

Comment:
`CreateFileVolumeUtilOld` ==> Rename this to `CreateFileVolumeUtilForVanillaK8s` or some other meaningful name

Reply:
We want to keep it old js to make to sure its an old implementation.

`VanillaK8s` doesn't use this code `CreateFileVolumeUtilForVanillaK8s` if auth check is enabled. so this will create confusion. Leave it as it.